src/backend/distributed/commands/table.c

```diff
@@ -1854,52 +1854,28 @@ PreprocessAlterTableSchemaStmt(Node *node, const char *queryString,
  * statement to be worked on the distributed table. Currently, it only processes
  * ALTER TABLE ... ADD FOREIGN KEY command to skip the validation step.
  */
-Node *
-SkipForeignKeyValidationIfConstraintIsFkey(AlterTableStmt *alterTableStatement,
-										   bool processLocalRelation)
+void
+SkipForeignKeyValidationIfConstraintIsFkey(AlterTableStmt *alterTableStatement)
 {
 	/* first check whether a distributed relation is affected */
 	if (alterTableStatement->relation == NULL)
 	{
-		return (Node *) alterTableStatement;
+		return;
 	}
 
 	LOCKMODE lockmode = AlterTableGetLockLevel(alterTableStatement->cmds);
 	Oid leftRelationId = AlterTableLookupRelation(alterTableStatement, lockmode);
 	if (!OidIsValid(leftRelationId))
 	{
-		return (Node *) alterTableStatement;
-	}
-
-	if (!IsCitusTable(leftRelationId) && !processLocalRelation)
-	{
-		return (Node *) alterTableStatement;
+		return;
 	}
 
-	/*
-	 * We check if there is a ADD FOREIGN CONSTRAINT command in sub commands list.
-	 * If there is we assign referenced releation id to rightRelationId and we also
-	 * set skip_validation to true to prevent PostgreSQL to verify validity of the
-	 * foreign constraint in master. Validity will be checked in workers anyway.
-	 */
-	List *commandList = alterTableStatement->cmds;
-	AlterTableCmd *command = NULL;
-	foreach_ptr(command, commandList)
+	if (!IsCitusTable(leftRelationId))
 	{
-		AlterTableType alterTableType = command->subtype;
-
-		if (alterTableType == AT_AddConstraint)
-		{
-			Constraint *constraint = (Constraint *) command->def;
-			if (constraint->contype == CONSTR_FOREIGN)
-			{
-				/* foreign constraint validations will be done in shards. */
-				constraint->skip_validation = true;
-			}
-		}
+		return;
 	}
 
-	return (Node *) alterTableStatement;
+	SkipConstraintValidation = true;
```

Comment:
better to use set_config_option here to make sure we roll back the value in case of error